src/Maps/MapData.h

```diff
@@ -14,16 +14,45 @@
 
 enum class CellType;
 
+namespace Stream {
+	class Writer;
+	class Reader;
+	class SeekableReader;
+}
+
 // FILE FORMAT DOCUMENTATION:
 //     Outpost2SVN\OllyDbg\InternalData\FileFormat SavedGame and Map.txt.
 
 // ALT IMPLEMENTATION (with COM support)
 //     Outpost2SVN\MapEditor\OP2Editor.
 
-//Represents the data stored in an Outpost 2 map file.
-struct MapData
+//An Outpost 2 map file.
+class MapData
 {
-	MapHeader header;
+public:
+	MapData();
+	MapData(const MapHeader& mapHeader);
+
+	MapHeader CreateHeader() const;
+
+	static void Write(const std::string& filename, const MapData& mapData);
+	static void Write(Stream::Writer& mapStream, const MapData& mapData);
+	static MapData ReadMap(std::string filename);
+	static MapData ReadMap(Stream::SeekableReader& seekableReader);
```

Comment:
Does processing a map really require a `SeekableReader`, or just a `Reader`? The saved game loading code skips over the saved game header, so that currently requires a `SeekableReader`, but that restriction doesn't apply to this function.

Reply:
Good point. I'll update shortly.